docs/docs/developers/tutorials/codealong/first_fees.md

```diff
@@ -0,0 +1,404 @@
+  const newWallet = await schnorrAccount.getWallet()
+  const feeJuiceReceipient = schnorrAccount.getAddress()
+
+  const claim = await feeJuicePortalManager.bridgeTokensPublic(feeJuiceReceipient, 1000000000000000000n, true);
+  // ...(wait or perform two txs)
+  const claimAndPay = new FeeJuicePaymentMethodWithClaim(newWallet, claim)
+  await schnorrAccount.deploy({ fee: { paymentMethod: claimAndPay }}).wait()
+
+```
+
+:::note  Payment: Fee Juice with claim from bridge
+Options for claim+pay with bridged funds that can be used in multiple commands:
+- CLI: `--from <sender address> --payment method=fee_juice,claim`
+- .js: `{ fee: { paymentMethod: new FeeJuicePaymentMethodWithClaim(newWallet, claim) }}`
+:::
+
+### Fee Paying Contract payment (public/private) - Advanced
+
+Setting up your own FPC and authorising the asset is an involved process outside the scope of this tutorial.
+Below we will look at the syntax for understanding.
+
+First register the FPC address in your pxe. In reality this might be an application funding users' transactions via their token.
+
+```bash
+aztec-wallet register-contract $FPC_ADDRESS FPCContract --from main
+aztec-wallet <command> --payment method=fpc-public,fpc=$FPC_ADDRESS,asset=$ASSET_ADDRESS
+```
+
+The second line can be any command that takes a `--payment` parameter. See `aztec-wallet --help` and the help of corresponding commands to check.
+
+Notice the specified address of both the fpc contract, and the asset address.
+
+The equivalent using aztec.js:
+
+```javascript
+import { FPCContract } from "@aztec/noir-contracts.js/FPC";
+  //...
+  const fpc = await FPCContract.deploy(wallets[0], bananaCoin.address, wallets[0].getAddress()).send().deployed();
+  // ... (setup token authorisation, and fpc balance here)
+  const publicFee = new PublicFeePaymentMethod(fpc.address, newWallet)
+
+```
+
+:::note  Payment: Fee Paying Contract (public/private)
+Options for payment via fpc that can be used in multiple commands:
+CLI: `--payment method=fpc-public,fpc=$FPC_ADDRESS,asset=$ASSET_ADDRESS` (`fpc-private` for private)
+.js: `{ fee: { paymentMethod: new PublicFeePaymentMethod(fpc.address, newWallet) }` (`PrivateFeePaymentMethod` for private)
+:::
+
+:::info Private fee payment
+Note: using a private fpc method is the only way for transactions to be paid for privately.
+
+Public and private refer to how the fpc will claim its tokens from the sender to then (publicly) spend fee juice. The visibility of the function being called is unchanged.
+:::
+
+## Summary of fee payment options
+
+The two key ways of paying for transactions: fee juice from an account or via an FPC. Both of which are contracts on the aztec network.
+
+### Fee juice from an account (default)
+
+- from the sender of a transaction (default)
+  - most common if making transactions from a funded account
+  - difficult to deploy without funds already
```

Comment:
```suggestion
  - requires bridging the fee asset from L1 to L2
```
not difficult, just requires an additional step

Reply:
And the preceding steps to acquire fee juice. This is making the point for devs that their users will need to acquire fee asset on L1 first, which requires an Ethereum wallet...
The best ux a dev can offer is some form of gating that funds their users on Aztec.

Reply:
Amended